aws-rds-dbinstance/aws-rds-dbinstance.json

```diff
@@ -0,0 +1,431 @@
+    },
+    "AllowMajorVersionUpgrade": {
+      "type": "boolean",
+      "description": "A value that indicates whether major version upgrades are allowed. Changing this parameter doesn't result in an outage and the change is asynchronously applied as soon as possible."
+    },
+    "AssociatedRoles": {
+      "type": "array",
+      "uniqueItems": true,
+      "items": {
+        "$ref": "#/definitions/DBInstanceRole"
+      },
+      "description": "The AWS Identity and Access Management (IAM) roles associated with the DB instance."
+    },
+    "AutoMinorVersionUpgrade": {
+      "type": "boolean",
+      "description": "A value that indicates whether minor engine upgrades are applied automatically to the DB instance during the maintenance window. By default, minor engine upgrades are applied automatically."
+    },
+    "AvailabilityZone": {
+      "type": "string",
+      "description": "The Availability Zone (AZ) where the database will be created. For information on AWS Regions and Availability Zones."
+    },
+    "BackupRetentionPeriod": {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 35,
+      "default": 1,
+      "description": "The number of days for which automated backups are retained. Setting this parameter to a positive number enables backups. Setting this parameter to 0 disables automated backups."
+    },
+    "CACertificateIdentifier": {
+      "type": "string",
+      "description": "The identifier of the CA certificate for this DB instance."
+    },
+    "CharacterSetName": {
+      "type": "string",
+      "description": "For supported engines, indicates that the DB instance should be associated with the specified character set."
+    },
+    "CopyTagsToSnapshot": {
+      "type": "boolean",
+      "description": "A value that indicates whether to copy tags from the DB instance to snapshots of the DB instance. By default, tags are not copied."
+    },
+    "DBClusterIdentifier": {
+      "type": "string",
+      "description": "The identifier of the DB cluster that the instance will belong to.",
+      "pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9]){0,62}$"
+    },
+    "DBInstanceClass": {
+      "type": "string",
+      "description": "The compute and memory capacity of the DB instance, for example, db.m4.large. Not all DB instance classes are available in all AWS Regions, or for all database engines."
+    },
+    "DBInstanceIdentifier": {
+      "type": "string",
+      "pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9]){0,62}$",
+      "description": "A name for the DB instance. If you specify a name, AWS CloudFormation converts it to lowercase. If you don't specify a name, AWS CloudFormation generates a unique physical ID and uses that ID for the DB instance."
+    },
+    "DBName": {
+      "type": "string",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9]{0,63}$",
+      "description": "The meaning of this parameter differs according to the database engine you use."
+    },
+    "DBParameterGroupName": {
```

Comment:
Should we add a pattern and minLength and maxLength? as It mentioned here in https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html as constrains under *DBParameterGroupName* like:
```
"pattern": "^[a-zA-Z]{1}(?:-?[a-zA-Z0-9])*$",
minLength: 1
maxLength: 255
```